reverse_engineering/helpers/utils.js

```diff
@@ -98,6 +98,42 @@ const getTemplateDocByJsonSchema = schema => {
 	}, {});
 };
 
+/**
+ *
+ * @param {Array<number | string | boolean>} list
+ * @returns {number | string | boolean}
+ */
+const getMostFrequentValueInList = list => {
+	const itemsToFrequencyMap = {};
+
+	list.forEach(item => {
+		if (itemsToFrequencyMap[item]) {
+			itemsToFrequencyMap[item] += 1;
+			return;
+		}
+
+		itemsToFrequencyMap[item] = 1;
+	});
+
+	const frequencyMapEntries = Object.entries(itemsToFrequencyMap);
+	const defaultMostFrequentEntry = frequencyMapEntries[0];
+
+	const mostFrequentItemEntry = frequencyMapEntries.reduce(
+		([mostFrequentItemValue, mostFrequentItemFrequency], [item, frequency]) => {
+			if (frequency > mostFrequentItemFrequency) {
+				return [item, frequency];
+			}
+
+			return [mostFrequentItemValue, mostFrequentItemFrequency];
+		},
+		defaultMostFrequentEntry,
+	);
+
+	const [mostFrequentItemValue] = mostFrequentItemEntry;
+
+	return mostFrequentItemValue;
```

Comment:
A small suggestion is to make it more readable. If you agree, please apply it.
```suggestion
	let mostFrequentItem = '';
    let maxFrequency = 0;

    for (let [item, frequency] of itemsToFrequencyMap) {
        if (frequency > maxFrequency) {
            mostFrequentItem = item;
            maxFrequency = frequency;
        }
    }

    return mostFrequentItem;
```